packages/langium/src/lsp/document-update-handler.ts

```diff
@@ -69,7 +72,10 @@ export class DefaultDocumentUpdateHandler implements DocumentUpdateHandler {
         }
     }
 
-    protected fireDocumentUpdate(changed: URI[], deleted: URI[]): void {
+    protected async fireDocumentUpdate(changed: URI[], deleted: URI[]): Promise<void> {
```

Comment:
When an async method is called in a synchronous context, `catch` blocks must be appended to avoid uncaught rejected promises. It might be easier to leave the method synchronous and use `then` with two callbacks in here.